src/main/java/org/skriptlang/skript/lang/experiment/ExperimentData.java

```diff
@@ -0,0 +1,219 @@
+	 * {@link #constructError()}.
+	 */
+	public String getErrorMessage()  {
+		return errorMessage;
+	}
+
+	/**
+	 * Check if the requirements of this {@link ExperimentData} are met.
+	 * @param experiments The current enabled {@link Experiment}s.
+	 * @return {@code True} if the requirements were met.
+	 */
+	public boolean checkRequirements(ExperimentSet experiments) {
+		if (!required.isEmpty()) {
+			for (Experiment experiment : required) {
+				if (!experiments.hasExperiment(experiment))
+					return false;
+			}
+		}
+		if (!disallowed.isEmpty()) {
+			for (Experiment experiment : disallowed) {
+				if (experiments.hasExperiment(experiment))
+					return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Check if the requirements of this {@link ExperimentData} are met.
+	 * If the requirements are not met, will produce a {@link Skript#error(String)} using {@link #errorMessage}.
+	 * @param experiments The current enabled {@link Experiment}s.
+	 * @return {@code True} if the requirements were met.
+	 */
+	public boolean checkRequirementsAndError(ExperimentSet experiments) {
+		if (!checkRequirements(experiments)) {
+			Skript.error(errorMessage);
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Construct a {@link String} combining what {@link Experiment}s need to be enabled and/or disabled in order to use.
+	 */
+	public String constructError() {
+		StringBuilder builder = new StringBuilder();
+		builder.append("This element is experimental. To use this, ");
+		if (!required.isEmpty()) {
+			builder.append("enable ");
+			builder.append(StringUtils.join(
+				required.stream()
+					.map(experiment -> "'" + experiment.codeName() + "'")
+					.toArray(),
+				", "));
+			if (!disallowed.isEmpty()) {
+				builder.append(" and disable ");
+				builder.append(StringUtils.join(
+					disallowed.stream()
+						.map(experiment -> "'" + experiment.codeName() + "'")
+						.toArray(),
+					", "));
+			}
+			builder.append(".");
+		} else {
+			assert !disallowed.isEmpty();
+			builder.append("disable ");
+			builder.append(StringUtils.join(
+				disallowed.stream()
+					.map(experiment -> "'" + experiment.codeName() + "'")
+					.toArray(),
+				", "));
+		}
+		return builder.toString();
+	}
+
+	public static class Builder {
+
+		private Set<Experiment> required = new HashSet<>();
+		private Set<Experiment> disallowed = new HashSet<>();
+		private @Nullable String errorMessage = null;
+
+		private Builder() {}
+
+		/**
+		 * Set the {@link Experiment}s that must be enabled in order to use.
+		 * @param required The {@link Experiment}s
+		 * @return This {@link Builder}.
+		 */
+		public Builder required(Experiment... required) {
+			this.required = Arrays.stream(required).collect(Collectors.toSet());
+			return this;
+		}
+
+		/**
+		 * Get the {@link Experiment}s that must be enabled in order to use.
+		 */
+		public Set<Experiment> required() {
```

Comment:
I don't think getters on the builder are necessary. You could make a `toBuilder()` on ExperimentData if you think it'd be useful.